From 275f4b82d992cecbab780ad6477d22d3d930cb6a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 08 Aug 2021 23:26:38 +0300
Subject: [PATCH] Revert "[SW-51761] gaudi: security: allow host sm access"

SM monitors are not needed to be accessed from PCIe and also are
vulnerable when accessed from PCIe and can result in various
instabilities. The same true for the gaps between the SM address space
and other components.
Given the above, we block the SM monitors and gaps access from PCIe.
Revert the commit that opens the full access and open it partially -
only monitors and gaps are blocked.

This reverts commit 67392f04247cb0e9d1ca37d755b835e2f8d1c61a.

Change-Id: Ic3d4caf27b8044905a1ce3202e28c14ed48b4272
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index 9cdb168..a708241 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -603,6 +603,8 @@
 }
 
 #define SM_DIST		0x20000
+#define SM_OBJS_MON_OFF	0x4000
+#define SM_OBJS_MON_SZ	0xC000
 #define SM_ENGINES	4
 static void preboot_secure_sm(void)
 {
@@ -1508,7 +1510,8 @@
 
 static void preboot_secure_lbw(void)
 {
-	u64 min, max;
+	int i;
+	u64 min, max, base;
 
 	/*
 	 * NOTE: The LBW and PCIe RRs HIT is: min < hit < max
@@ -1629,6 +1632,15 @@
 	 */
 	max = mmTPC7_EML_CS_BASE + TPC7_EML_CS_MAX_OFFSET - 1;
 	preboot_secure_lbw_rr_if(LBW_RR_DBG_ETR, min, max);
+
+	/* SYNC_MNGR: PCIe only! PCIe RR_26 up to PCIe RR_29 */
+	for (i = 0; i < 4; i++) {
+		base = mmSYNC_MNGR_GLBL_W_S_BASE + SM_OBJS_MON_OFF;
+		base += SM_DIST * i;
+		min = base - 1;
+		max = base + SM_OBJS_MON_SZ;
+		preboot_secure_pcie_rr(26 + i, min, max);
+	}
 }
 
 static void preboot_hbw_rr_set_rtr_wr(int rr, int rtr, u64 base, u64 mask,
@@ -1801,7 +1813,7 @@
 	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
 	 * STM, SPI Flash, Scratchpad SRAM, PCIe FW SRAM.
 	 * SPSRAM space for fit is used in dynamic RR.
-	 * Assign the PCIe IF RR_16, RR_22, RR_24 (0-15 are allocated
+	 * Assign the PCIe IF RR_16, RR_22, RR_24 (0-15 and 26-31 are allocated
 	 * for LBW) for this job.
 	 * NOTE: The PCIe RRs use min/max concept and HIT is: min < hit < max
 	 */
